### DISCUSSION TITLE
Failed to create decryptor: Incorrect AES key length (48 bytes)

### DISCUSSION THREAD

**Discussion by johnnyzilevu47** [streamlink/streamlink#4870] [Q&A] [answered]
I dont know whats happening with this url
https://www.futurly.com/s/preview/courses/blender-architecture#6337f80fe4b05a534f0e3d5a

#EXT-X-KEY:METHOD=AES-128,URI="k/timestamp",IV=0x496daa1c6914000e408c65cead91fc29
it says k/timestamp 

streamlink error - [stream.hls][error] Failed to create decryptor: Incorrect AES key length (48 bytes)

i dont know how to solve this , can anyone pls guide me on this

**Comment by bastimeyer** [comment, marked as answer]
> it says k/timestamp

The site is using a custom HLS implementation for retrieving the key. If the key URL is static, you can use `--hls-segment-key-uri` to override the key URI
https://streamlink.github.io/cli.html#cmdoption-hls-segment-key-uri

but if more is required for acquiring the key, then you'll have to write a custom plugin with a custom HLS implementation where you'll need to subclass the `HLSStreamWriter` and handle the key acquirement appropriately.

**Reply by bastimeyer** [reply]
The `file://...` scheme is also supported by `--hls-segment-key-uri`, in case you don't want to write a plugin and instead custom external key-retrieval logic.

**Reply by johnnyzilevu47** [reply]
I am not a programmer so i won't be able to write custom plugin , can you please throw some hints on how exactly i do this to get the key url 
From URI="k/timestamp

**Reply by bastimeyer** [reply]
Depends on the stream you're trying to watch. You'll have to figure this out on your own. Take a look at your browser's dev tools. Good luck.
